### ISSUE TITLE
Publish multi-jvm cluster helpers (i.e. MultiNodeClusterSpec) as a dedicated package.

### ISSUE THREAD

**Issue by e-kolpakov** [akka/akka#23945]
Akka cluster docs contain some instructions on [testing cluster applications](https://doc.akka.io/docs/akka/2.5/scala/cluster-usage.html#how-to-test), but the apporach suggested there is very barebone as it basically requires managing cluster state by hand.

However, akka-cluster [multi-jvm tests](https://github.com/akka/akka/tree/master/akka-cluster/src/multi-jvm/scala/akka/cluster) contain a much more sophisticated test helpers that allow:

* Waiting for the [cluster to be in "Up" state](https://github.com/akka/akka/blob/master/akka-cluster/src/multi-jvm/scala/akka/cluster/MultiNodeClusterSpec.scala#L216)
* Assert that [leader is elected](https://github.com/akka/akka/blob/master/akka-cluster/src/multi-jvm/scala/akka/cluster/MultiNodeClusterSpec.scala#L284)
* Await for [members to join and reach "Up" state](https://github.com/akka/akka/blob/master/akka-cluster/src/multi-jvm/scala/akka/cluster/MultiNodeClusterSpec.scala#L300)
* ... and much more.

Unfortunately all that goodness is locked in a package that is not published. It would be great to have access to such tools as part of `akka-multi-node-testkit` or a dedicated package.

**Comment by patriknw** [comment]
Most of this can be done with awaitAssert and cluster.state, so I’m not sure it’s needed to create abstractions on top of that. The documentation can probably be improved though, with examples for these typical things.

**Comment by e-kolpakov** [comment]
@packtpartner ok, let me rephrase this than - does it make sense to include helpers for typical scenarios into the testkit? I believe it would be a net loss to everyone to reimplement "cluster up" and "member joined" checks using cluster state.

Also, I might be able to allocate some time for submitting such helpers as open-source pull request, but I'd like to get green light for that feature before starting to work on it. Is this a good place to discuss it, or should I use something else (JIRA/gitter channel/etc.)?

**Comment by patriknw** [comment]
Give it a try to write a few such helpers in a test in akka-cluster/src/test/scala and I’ll judge if they deserve a new akka-cluster-testkit module or should just be included as snippets in docs.

**Comment by sergiy0** [comment]
I'm also considering `MultiNodeClusterSpec` as very helpful and voting for publishing it in some way.
